tapeout/src/main/scala/transforms/macros/Utils.scala

```diff
@@ -0,0 +1,154 @@
+// See LICENSE for license details.
+
+package barstools.tapeout.transforms.macros
+
+import firrtl._
+import firrtl.ir._
+import firrtl.PrimOps
+import firrtl.Utils.{ceilLog2, BoolType}
+import scala.util.parsing.json.JSON // Todo: this will be gone
+import java.io.File
+import scala.language.implicitConversions
+
+trait PortPolarity
+case object ActiveLow extends PortPolarity
+case object ActiveHigh extends PortPolarity
+case object NegativeEdge extends PortPolarity
+case object PositiveEdge extends PortPolarity
+object PortPolarity {
+  implicit def toPortPolarity(s: Any): PortPolarity =
+    (s: @unchecked) match {
+      case "active low" => ActiveLow
+      case "active high" => ActiveHigh
+      case "negative edge" => NegativeEdge 
+      case "positive edge" => PositiveEdge
+    }
+  implicit def toPortPolarity(s: Option[Any]): Option[PortPolarity] =
+    s map toPortPolarity
+}
+
+case class MacroPort(
+    clockName: String,
+    clockPolarity: Option[PortPolarity],
+    addressName: String,
+    addressPolarity: Option[PortPolarity],
+    inputName: Option[String],
+    inputPolarity: Option[PortPolarity],
+    outputName: Option[String],
+    outputPolarity: Option[PortPolarity],
+    chipEnableName: Option[String],
+    chipEnablePolarity: Option[PortPolarity],
+    readEnableName: Option[String],
+    readEnablePolarity: Option[PortPolarity],
+    writeEnableName: Option[String],
+    writeEnablePolarity: Option[PortPolarity],
+    maskName: Option[String],
+    maskPolarity: Option[PortPolarity],
+    maskGran: Option[BigInt],
+    width: BigInt,
+    depth: BigInt) {
+  val effectiveMaskGran = maskGran.getOrElse(width)
+  val AddrType = UIntType(IntWidth(ceilLog2(depth) max 1))
+  val DataType = UIntType(IntWidth(width))
+  val MaskType = UIntType(IntWidth(width / effectiveMaskGran))
```

Comment:
Yeah, I just want to make them consistent for types. I'm also ok to change them.